File: AgoraChat-ObjectiveC/AgoraChat/Code/Custom/ViewController/ACDContainerSearchTableViewController.m
//
//  ACDContainerSearchTableViewController.m
//  ChatDemo-UI3.0
//
//  Created by liang on 2021/10/29.
//  Copyright © 2021 easemob. All rights reserved.
//

#import "ACDContainerSearchTableViewController.h"
#import "MISScrollPage.h"

@interface ACDContainerSearchTableViewController ()<MISScrollPageControllerContentSubViewControllerDelegate>

@end

@implementation ACDContainerSearchTableViewController

- (void)viewDidLoad {
    [super viewDidLoad];
}


#pragma mark - MISScrollPageControllerContentSubViewControllerDelegate
- (BOOL)hasAlreadyLoaded{
    return NO;
}

- (void)viewDidLoadedForIndex:(NSUInteger)index{
    
}

- (void)viewWillAppearForIndex:(NSUInteger)index{
    [self cancelSearchState];

}

- (void)viewDidAppearForIndex:(NSUInteger)index{
}

- (void)viewWillDisappearForIndex:(NSUInteger)index{
    self.editing = NO;
    [self cancelSearchState];
}

- (void)viewDidDisappearForIndex:(NSUInteger)index{
    
}


@end

File: AgoraChat-Swift/AgoraChat-Swift/code/Custom/ViewController/ACDContainerSearchTableViewController.swift
//
//  ACDContainerSearchTableViewController.swift
//  AgoraChat-Swift
//
//  Created by 冯钊 on 2022/10/26.
//

import UIKit
import MISScrollPage

class ACDContainerSearchTableViewController<T>: ACDSearchTableViewController<T>, MISScrollPageControllerContentSubViewControllerDelegate {

    func hasAlreadyLoaded() -> Bool {
        return false
    }

    func viewWillAppear(for index: UInt) {
        self.cancelSearchState()
    }
    
    func viewWillDisappear(for index: UInt) {
        self.isEditing = false
        self.cancelSearchState()
    }
}
